java/src/main/native/src/TableJni.cpp

```diff
@@ -245,6 +248,49 @@ class native_arrow_ipc_writer_handle final {
     }
     initialized = false;
   }
+
+  std::vector<cudf::column_metadata> get_column_metadata(const table_view& tview) {
+    if (!column_names.empty() && columns_meta.empty()) {
+      // Rebuild the structure of column meta according to table schema.
+      // All the tables written by this writer should share the same schema,
+      // so build column metadata only once.
+      columns_meta.reserve(tview.num_columns());
+      size_t idx = 0;
+      for (auto itr = tview.begin(); itr < tview.end(); ++itr) {
+        columns_meta.push_back(std::move(build_one_column_meta(*itr, idx)));
+        idx ++;
+      }
+    }
+    return columns_meta;
+  }
+
+private:
+  // Still return an oject instead of being passed as an out argument, even
+  // `column_metadata` has no move constructor and would be copied.
```

Comment:
I don't think this is true.  The compiler should be generating an implicit move constructor for the class.  https://en.cppreference.com/w/cpp/language/move_constructor

Reply:
good to know, thanks